avl_tree/AVL_Tree.java

```diff
@@ -0,0 +1,294 @@
+				avl.del(d);
+				break;
+			case 3:
+				if(avl.isEmpty()){
+					System.out.println("null");
+				}
+				else{
+					avl.prntIn(avl.root);
+					System.out.println(" ");
+				}
+				break;
+			case 4:
+				break;
+			default:
+				System.out.println("Sorry input number is not valid ");
+				break;
+			}
+			if (a == 4) {
+				break;
+			}
+		}
+	}
+}
+
+class AVL {
+	public NodeAVL root = null;
+
+	public void insert(int data) {
+		if (root == null) {
+			root = new NodeAVL(data);
+		} else
+			root = insert(root, data);
+	}
+
+	private NodeAVL insert(NodeAVL node, int data) {
+		if (node == null) {
+			node = new NodeAVL(data);
+			return node;
+		} else {
+			if (node.data > data) {
+				node.left = insert(node.left, data);
+				if (node.left == null)
+					node.hLeft = 0;
+				else
+					node.hLeft = Math.max(node.left.hLeft, node.left.hRight) + 1;
+			} else {
+				node.right = insert(node.right, data);
+				if (node.right == null)
+					node.hRight = 0;
+				else
+					node.hRight = Math.max(node.right.hLeft, node.right.hRight) + 1;
+			}
+			node = isRotate(node);
+		}
+		return node;
+	}
+
+	public NodeAVL rotateLR(NodeAVL node) {
+		NodeAVL sec = node.left;
+		NodeAVL temp = sec.right;
+		node.left = temp;
+		sec.right = temp.left;
+		temp.left = sec;
+		node.left = temp.right;
+		temp.right = node;
+		if (node.left == null)
+			node.hLeft = 0;
+		else
+			node.hLeft = Math.max(node.left.hLeft, node.left.hRight) + 1;
+		if (sec.right == null)
+			sec.hRight = 0;
+		else
+			sec.hRight = Math.max(sec.right.hLeft, sec.right.hRight) + 1;
+		temp.hLeft = Math.max(sec.hLeft, sec.hRight) + 1;
+		temp.hRight = Math.max(node.hLeft, node.hRight) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateRL(NodeAVL node) {
+		NodeAVL sec = node.right;
+		NodeAVL temp = sec.left;
+		node.right = temp;
+		sec.left = temp.right;
+		temp.right = sec;
+		node.right = temp.left;
+		temp.left = node;
+		if (node.right == null)
+			node.hRight = 0;
+		else
+			node.hRight = Math.max(node.right.hLeft, node.right.hRight) + 1;
+		if (sec.left == null)
+			sec.hLeft = 0;
+		else
+			sec.hLeft = Math.max(sec.left.hLeft, sec.left.hRight) + 1;
+		temp.hRight = Math.max(sec.hLeft, sec.hRight) + 1;
+		temp.hLeft = Math.max(node.hLeft, node.hRight) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateLL(NodeAVL node) {
+		NodeAVL temp = node.left;
+		node.left = temp.right;
+		temp.right = node;
+		if (node.left == null)
+			node.hLeft = 0;
+		else
+			node.hLeft = Math.max(node.left.hRight, node.left.hLeft) + 1;
+		temp.hRight = Math.max(node.hRight, node.hLeft) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateRR(NodeAVL node) {
+		NodeAVL temp = node.right;
+		node.right = temp.left;
+		temp.left = node;
+		if (node.right == null)
+			node.hRight = 0;
+		else
+			node.hRight = Math.max(node.right.hRight, node.right.hLeft) + 1;
+		temp.hLeft = Math.max(node.hRight, node.hLeft) + 1;
+		return temp;
+	}
+
+	public NodeAVL isRotate(NodeAVL node) {

```

Comment:
`isRotate` seems kind of unclear in its name, either give it some informative name or add some comments about it.